tests/test_dependencies.py

```diff
@@ -472,6 +393,581 @@ class ComponentWithScript(SimpleComponent):
             ComponentWithScript.render(kwargs={"variable": "foo"})
 
 
+@djc_test
+class TestDependenciesStrategyDocument:
+    def test_inserts_styles_and_script_to_default_places_if_not_overriden(self):
+        registry.register(name="test", component=SimpleComponent)
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            <!DOCTYPE html>
+            <html>
+                <head></head>
+                <body>
+                    {% component "test" variable="foo" / %}
+                </body>
+            </html>
+        """
+        rendered_raw = Template(template_str).render(Context({}))
+        rendered = render_dependencies(rendered_raw)
```

Comment:
I think I would set dep_strategy="document" here just to be explicit

Reply:
Thanks, fixed!